packages/plugin-ext/src/main/browser/menus/menus-contribution-handler.ts

```diff
@@ -98,154 +98,123 @@ export class MenusContributionPointHandler {
         if (!allMenus) {
             return Disposable.NULL;
         }
-        const allSubmenus = plugin.contributes && plugin.contributes.submenus;
         const toDispose = new DisposableCollection();
-        for (const location in allMenus) {
-            if (location === 'commandPalette') {
-                for (const menu of allMenus[location]) {
-                    if (menu.command && menu.when) {
-                        toDispose.push(this.quickCommandService.pushCommandContext(menu.command, menu.when));
-                    }
-                }
-            } else if (location === 'editor/title') {
-                for (const action of allMenus[location]) {
-                    if (!action.command) {
-                        continue;
-                    }
-                    toDispose.push(this.registerTitleAction(location, action, {
-                        execute: widget => this.codeEditorWidgetUtil.is(widget) && this.commands.executeCommand(action.command!, this.codeEditorWidgetUtil.getResourceUri(widget)),
-                        isEnabled: widget => this.codeEditorWidgetUtil.is(widget) && this.commands.isEnabled(action.command!, this.codeEditorWidgetUtil.getResourceUri(widget)),
-                        isVisible: widget => this.codeEditorWidgetUtil.is(widget) && this.commands.isVisible(action.command!, this.codeEditorWidgetUtil.getResourceUri(widget))
-                    }));
-                }
-            } else if (location === 'view/title') {
-                for (const action of allMenus[location]) {
-                    if (!action.command) {
-                        continue;
+
+        const tree = this.getMenusTree(plugin);
+        tree.forEach(rootMenu => {
+
+            const registerMenuActions = (menus: MenuTree[], group: string | undefined, submenusOrder: string | undefined = '') => {
+                menus.forEach(menu => {
+                    if (group) {
+                        // Adding previous group to the start of current menu group.
+                        menu.group = `${group}/${menu.group || '_'}`;
```

Comment:
Also updated default value for group from `''` to `'_'` (If there's no group the item should be at the end of the menu tree - compared with VSCode)